### PR TITLE
feat(presenter): remove auth service

### PR DESCRIPTION
**Pull request by Mogge**
## 🍰 Pullrequest

 Based on #404 

### Issues

- None

### Todo

- [X] None

### CODE REVIEW

**Review by Elweyn — CHANGES_REQUESTED** [review]
There is a Problem in the following szenario:
- Going on presenter localhost:3001
- Sign in
- redirect to frontend localhost:3000
- call localhost:3001
- go back to localhost:3000 here we get redirected to localhost:3001 as if we are not signed in

**Comment by Mogge** [comment]
This is the behavior on master. Has nothing to do with this PR.

**Comment by Mogge** [comment]
Furthermore I think that this behavior is correct

File: presenter/.env.dist [hunk 1]
```diff
@@ -1,15 +1,6 @@
 # AUTH
-# PUBLIC_ENV__AUTH__AUTHORITY and PUBLIC_ENV__AUTH__AUTHORITY_SIGNUP_URI
-# are not default values since the default is to hide the signin/signup buttons
 PUBLIC_ENV__AUTH__AUTHORITY="http://localhost:9000/application/o/dreammallearth/"
```
Comment (ulfgebhardt) [inline presenter/.env.dist:2]: what we need this for?

```suggestion
```

File: presenter/.env.dist [hunk 2]
```diff
@@ -1,15 +1,6 @@
...
 PUBLIC_ENV__AUTH__AUTHORITY_SIGNUP_URI="http://localhost:9000/if/flow/dreammallearth-enrollment/"
-PUBLIC_ENV__AUTH__AUTHORITY_SIGNOUT_URI="http://localhost:9000/if/flow/dreammallearth-invalidation-flow/"
-PUBLIC_ENV__AUTH__CLIENT_ID="G3g0sjCjph1NAyGeeu5Te5ltx1I7WZ0DGB8i6vOI"
-PUBLIC_ENV__AUTH__REDIRECT_URI="http://localhost:3000/auth"
-PUBLIC_ENV__AUTH__SILENT_REDIRECT_URI="http://localhost:3000/silent-refresh"
-PUBLIC_ENV__AUTH__RESPONSE_TYPE="code"
-PUBLIC_ENV__AUTH__SCOPE="openid profile posts"
-
 PUBLIC_ENV__SIGNIN_REDIRECT_URI="http://localhost:3000/signin"
```
Comment (ulfgebhardt) [inline presenter/.env.dist:4]: i'd like if we can find a naming convention for both redirect urls

File: presenter/src/components/menu/HeaderMenu.vue [hunk 3]
```diff
@@ -139,44 +113,16 @@ import LogoImage from '#components/LogoImage.vue'
 import AnchorLink from '#components/nav/AnchorLink.vue'
 // import { querySecret } from '#queries/querySecret'
 import { AUTH } from '#src/env'
-import AuthService from '#src/services/AuthService'
-import { useAuthStore } from '#stores/authStore'
 // import { ApolloClient, InMemoryCache } from '@apollo/client/core'
 
-const authService = inject<AuthService>('authService')
-const auth = useAuthStore()
 const showAuthentication = AUTH.AUTHORITY && AUTH.AUTHORITY_SIGNUP_URI
```
Comment (ulfgebhardt) [inline presenter/src/components/menu/HeaderMenu.vue:118]:
```suggestion
const showAuthentication = AUTH.AUTHORITY_SIGNUP_URI
```

File: presenter/src/components/menu/HeaderMenu.vue [hunk 4]
```diff
@@ -97,16 +80,7 @@
       </div>
       <div v-if="showAuthentication" class="">
```
Comment (ulfgebhardt) [inline presenter/src/components/menu/HeaderMenu.vue:81]: We must show the correpsonding button only when the url is given. If the url is missing, dont show the button. There should be two checks accordingly.

File: presenter/src/env.ts [hunk 5]
```diff
@@ -1,18 +1,8 @@
 const AUTH = {
   AUTHORITY: (import.meta.env.PUBLIC_ENV__AUTH__AUTHORITY ?? '') as string,
```
Comment (ulfgebhardt) [inline presenter/src/env.ts:2]: remove
```suggestion
```

File: presenter/src/env.ts [hunk 6]
```diff
@@ -1,18 +1,8 @@
...
-  AUTHORITY_SIGNOUT_URI: (import.meta.env.PUBLIC_ENV__AUTH__AUTHORITY_SIGNOUT_URI ??
-    'http://localhost:9000/if/flow/dreammallearth-invalidation-flow/') as string,
-  CLIENT_ID: (import.meta.env.PUBLIC_ENV__AUTH__CLIENT_ID ??
-    'G3g0sjCjph1NAyGeeu5Te5ltx1I7WZ0DGB8i6vOI') as string,
-  REDIRECT_URI: (import.meta.env.PUBLIC_ENV__AUTH__REDIRECT_URI ??
-    'http://localhost:3000/auth') as string,
-  SILENT_REDIRECT_URI: (import.meta.env.PUBLIC_ENV__AUTH__SILENT_REDIRECT_URI ??
-    'http://localhost:3000/silent-refresh') as string,
-  RESPONSE_TYPE: (import.meta.env.PUBLIC_ENV__AUTH__RESPONSE_TYPE ?? 'code') as string,
   SIGNIN_REDIRECT_URI: (import.meta.env.PUBLIC_ENV__SIGNIN_REDIRECT_URI ??
```
Comment (ulfgebhardt) [inline presenter/src/env.ts:4]: unify naming

File: presenter/.env.dist [hunk 7]
```diff
@@ -1,15 +1,5 @@
...
-PUBLIC_ENV__AUTH__AUTHORITY_SIGNUP_URI="http://localhost:9000/if/flow/dreammallearth-enrollment/"
-PUBLIC_ENV__AUTH__AUTHORITY_SIGNOUT_URI="http://localhost:9000/if/flow/dreammallearth-invalidation-flow/"
-PUBLIC_ENV__AUTH__CLIENT_ID="G3g0sjCjph1NAyGeeu5Te5ltx1I7WZ0DGB8i6vOI"
-PUBLIC_ENV__AUTH__REDIRECT_URI="http://localhost:3000/auth"
-PUBLIC_ENV__AUTH__SILENT_REDIRECT_URI="http://localhost:3000/silent-refresh"
-PUBLIC_ENV__AUTH__RESPONSE_TYPE="code"
-PUBLIC_ENV__AUTH__SCOPE="openid profile posts"
-
+PUBLIC_ENV__SIGNUP_URI="http://localhost:9000/if/flow/dreammallearth-enrollment/"
 PUBLIC_ENV__SIGNIN_REDIRECT_URI="http://localhost:3000/signin"
```
Comment (ulfgebhardt) [inline presenter/.env.dist:3]: is this correct? You reference `SIGNIN_URI` not `SIGNIN_REDIRECT_URI` below

Comment (Mogge) [reply presenter/.env.dist]: Good catch!

**Review by ulfgebhardt — APPROVED** [review]

**Pull request merged**